src/lower/test.rs

```diff
@@ -191,3 +225,126 @@ fn check_parameter_kinds() {
         }
     }
 }
+
+#[test]
+fn two_impls_for_same_type() {
+    lowering_error! {
+        program {
+            trait Foo { }
+            struct Bar { }
+            impl Foo for Bar { }
+            impl Foo for Bar { }
+        }
+        error_msg {
+            "overlapping impls of trait \"Foo\""
+        }
+    }
+}
+
+#[test]
+fn generic_vec_and_specific_vec() {
+    lowering_error! {
+        program {
+            trait Foo { }
+            struct Vec<T> { }
+            struct Bar { }
+            impl Foo for Vec<Bar> { }
+            impl<T> Foo for Vec<T> { }
+        }
+        error_msg {
+            "overlapping impls of trait \"Foo\""
+        }
+    }
+}
+
+#[test]
+fn concrete_impl_and_blanket_impl() {
+    lowering_error! {
+        program {
+            trait Foo { }
+            struct Bar { }
+            impl Foo for Bar { }
+            impl<T> Foo for T { }
+        }
+        error_msg {
+            "overlapping impls of trait \"Foo\""
+        }
+    }
+}
+
+#[test]
+fn two_blanket_impls() {
+    lowering_error! {
+        program {
+            trait Foo { }
+            trait Bar { }
+            trait Baz { }
+            impl<T> Foo for T where T: Bar { }
+            impl<T> Foo for T where T: Baz { }
+            struct Quux { }
+            impl Bar for Quux { }
+            impl Baz for Quux { }
+        }
+        error_msg {
+            "overlapping impls of trait \"Foo\""
+        }
+    }
+}
+
+#[test]
+#[ignore] // Ignored until we have the compat modality
```

Comment:
rather than use `ignore`, I would prefer you write the test so that it tests the current behavior, but leave a comment explaining how it's wrong. This way we're more likely to notice when you fix it and re-enable the test...